org.eclipse.dawnsci.nexus.test/src/org/eclipse/dawnsci/nexus/ComplexNexusFileTest.java

```diff
@@ -54,13 +55,13 @@ private void setupTestData() {
 		testData.put("actualTime", createDoubleDataset(
 				1.39112723398E9, 1.391127234478E9, 1.391127234673E9, 1.391127234871E9, 1.391127235068E9,
 				1.391127248978E9, 1.391127249481E9, 1.391127249751E9, 1.391127249953E9, 1.391127250154E9));
-		testData.put("beamok", DatasetFactory.ones(shape, Dataset.FLOAT64));
+		testData.put("beamok", DatasetFactory.ones(shape));
 		testData.put("ionc_i", createDoubleDataset(
 				8.374020580781096, 8.370683990350841, 8.370854377232897, 8.373055360469023, 8.375409630828424,
 				8.363466060032364, 8.361367064643199, 8.360248862560692, 8.358109255244436, 8.358963327124483));
 		testData.put("count_time", createFilledDataset(0.1, size));
 		testData.put("data", createDetectorDataDataset());
-		testData.put("region_origin", DatasetFactory.zeros(new int[] { 1,  2 }, Dataset.FLOAT32));
+		testData.put("region_origin", DatasetFactory.zeros(FloatDataset.class, 1,  2));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "region_origin" 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)